### PR TITLE
Events solution

### PR DESCRIPTION
**Pull request by tverdohleb**

### CODE REVIEW

File: eventListener.js [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+        this.subscribeToNewBlockHeaders = this.subscribeToNewBlockHeaders.bind(this);
+        this.subscribeToSyncing = this.subscribeToSyncing.bind(this);
+        this.unSubscribe = this.unSubscribe.bind(this);
+    }
+
+    setEvent (type, params, cb, eventEmitter) {
+      if (!this.web3Obj) {
+        const web3Provider = new Web3(new Web3.providers.WebsocketProvider('wss://ropsten.infura.io/ws'));
+        this.web3Obj = new Web3(web3Provider);
+      }
```
Comment (jjperezaguinaga) [inline eventListener.js]: This doesn't seem correct, you are instantiating `web3Provider` with `new Web3` and then calling it again.

Furthermore, you can do this in one line:
```
this.web3Instance = !this.web3Instance && new Web3(new Web3.providers.WebsocketProvider('wss://ropsten.infura.io/ws'));
```

File: eventListener.js [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+        if (err) console.log(err);
+      }
+
+      let eventSubscription;
+
+      if(params) {
+          eventSubscription =  this.web3Obj.eth.subscribe(type, params, errCb);
+      } else {
+        eventSubscription = this.web3Obj.eth.subscribe(type, errCb)
+      }
```
Comment (jjperezaguinaga) [inline eventListener.js]: 7 lines of code to do the following:
```
const eventSubscription = params ? this.web3Obj.eth.subscribe(type, params, errCb) : this.web3Obj.eth.subscribe(type, errCb)
```
Please try to keep your code dryer.

File: index.js [hunk 3]
```diff
@@ -1,7 +1,9 @@
 const ContractArtifacts = require("@mybit/contracts");
-const Chain = require("@mybit/chain");
+// const Chain = require("@mybit/chain");
```
Comment (jjperezaguinaga) [inline index.js]: Just remove it, no point in commenting it when we have version control.

File: index.js [hunk 4]
```diff
@@ -56,43 +61,43 @@ module.exports = (function (){
 
   return {
     api: async () => {
-      return await apiContract.at(Chain.API());
+      return await apiContract.at('0x7c728214be9a0049e6a86f2137ec61030d0aa964');
```
Comment (jjperezaguinaga) [inline index.js]: These are _not_ hardcoded. They might change, which is the reason why we are calling them externally. These should be called from `@mybit/contracts`, not `@mybit/chain`, that's part of the issue, but we do not hardcode them.

File: index.js [hunk 5]
```diff
@@ -56,43 +61,43 @@ module.exports = (function (){
...
-      return await assetGeneratorContract.at(Chain.AssetGenerator());
+      return await assetGeneratorContract.at('0xf16165f1046f1b3cdb37da25e835b986e696313a');
     },
 
-    assetManagerEscrow: async () => {
-      return await assetManagerEscrowContract.at(Chain.BrokerEscrow());
-    },
+    // assetManagerEscrow: async () => {
+    //   return await assetManagerEscrowContract.at(Chain.BrokerEscrow());
+    // },
```
Comment (jjperezaguinaga) [inline index.js]: Either remove them or leave them. In this case, why we are removing this?

Comment (tverdohleb) [reply index.js]: because I didn`t find BrokerEscrow address in Chain.js

**Pull request closed without merging**